### PR TITLE
TaxedToken implementation that charges a fee for token transfers. 

### LINKED ISSUE #787: TaxedToken implementation that charges a fee for token transfers
## 🎉 Description

An interesting erc20 token use-case that I keep seeing is a 'taxed' transfer (i.e. a small fee (for example 1%) is diverted to a `feeAccount` in `transfer`). I implemented this in a `TaxedToken.sol` contract w/ pretty thorough unit-tests. Would there be any interest in a PR w/ this functionality?

- [ ] 🐛 This is a bug report.
- [x] 📈 This is a feature request.

## 🔢 Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

```solidity
  /**
   * @dev Transfer tokens to a specified address after diverting a fee to a central account.
   * @param _to The receiving address.
   * @param _value The number of tokens to transfer.
   */
  function transfer(address _to, uint256 _value) public returns (bool) {
    require(_to != address(0));
    require(_value <= balances[msg.sender]);
    require(_value % (uint256(10) ** decimals) == 0);

    uint fee = SafeMath.div(SafeMath.mul(_value, transferFeePercentage), 100);
    uint taxedValue = SafeMath.sub(_value, fee);

    // SafeMath.sub will throw if there is not enough balance.
    balances[msg.sender] = SafeMath.sub(balances[msg.sender], _value);

    balances[_to] = SafeMath.add(balances[_to], taxedValue);
    Transfer(msg.sender, _to, taxedValue);
    balances[feeAccount] = SafeMath.add(balances[feeAccount], fee);
    Transfer(msg.sender, feeAccount, fee);

    return true;
  }
```

## 👍 Other Information

https://ethereum.stackexchange.com/questions/30957/is-an-additional-transaction-fee-possible-in-the-erc20-token?rq=1

### PR DESCRIPTION
**Pull request by mjdietzx**
Fixes #787 

# 🚀 Description

An interesting erc20 token use-case I keep seeing is a 'taxed' transfer (i.e. a small fee (for example 1%) is diverted to a feeAccount in `transfer`). This PR implements that functionality and includes corresponding unit-tests.

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](/docs/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: contracts/token/ERC20/TaxedToken.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   * @param _value The number of tokens to transfer.
+   */
+  function transfer(address _to, uint256 _value) public returns (bool) {
+    require(_to != address(0));
+    require(_value <= balances[msg.sender]);
+    require(_value % (uint256(10) ** decimals) == 0);
+
+    balances[msg.sender] = balances[msg.sender].sub(_value);
+
+    uint256 fee = Math.min256(_value.mul(transferFeePercentage).div(100), maxTransferFee);
```
Comment (Shrugs) [inline contracts/token/ERC20/TaxedToken.sol]: is it possible to use `super.transfer()` here, to avoid reimplementing lines 23, 24, 27, 32, 33 and 38?

Comment (mjdietzx) [reply contracts/token/ERC20/TaxedToken.sol]: yes, good point! I think this is possible and is the cleaner solution. I will update the PR w/ improvement

**Comment by Shrugs** [comment]
LGTM, but haven't done a thorough look-through of the code to make sure all the math works, etc. Requesting a review from a fellow maintainer :)

**Comment by nventuro** [comment]
Closing due to staleness. Also, this would be better implemented in the current version of OpenZeppelin by overriding the `_transfer` function.

**Pull request closed without merging**